Overview.bs

```diff
@@ -260,6 +269,175 @@ significant bit in the byte and the bit with the largest index is the most signi
+<tr><td>2</td><td>4</td></tr>
+<tr><td>3</td><td>6</td></tr>
+<tr><td>4</td><td>8</td></tr>
+<tr><td>-1</td><td>1</td></tr>
+<tr><td>-2</td><td>3</td></tr>
+<tr><td>-3</td><td>5</td></tr>
+<tr><td>-4</td><td>7</td></tr>
+</table>
+</div>
+
+Psuedo code:
+<figure>
+```
+encode(n) {
+  if (n >= 0)
+    return n * 2
+  else
+    return (n * -2) - 1
+}
+
+decode(n) {
+  if (n & 1)
+    return -((n + 1) / 2)
+  else
+    return n / 2
+}
+```
+
+<div class="example">
+```
+delta_list = [23, 20, -31, -9, 64, 1, 1, -69]
+zig_zag_encoded_list = [46, 40, 61, 17, 128, 2, 2, 137]
+```
+</div>
+
+<figcaption>Psuedo-code for Zig-Zag encoding and decoding</figcaption>
+</figure>
+
+*Add reference to proto implementation*
+
+
+#### Variable Length Integer (varint) Encoding #### {#compressedintlist-varint}
+
+Unsigned integers are encoded as a sequence of bytes. Small values require one
+byte, larger values require two bytes etc... 7 bits of the value at a time are encoded.
+These 7 bits are place in the lower 7 bits of the output byte. The high bit is
+used to indicate if additional bytes follow.
+
+The decoder reads and processes bytes while the high bit is set.
+
+The values 0..127 are thus encoded as a singe byte, with the high bit clear.
+The values 128..16383 require 2 bytes. At most 5 bytes are required to encoded
+the full 64 bit unsigned range.
+
+Examples of the encoding:
+
+<div class="example">
+<table>
+<tr><th>Value</th><th>Value (binary)</th><th>Output Bytes</th></tr>
+<tr><td>0</td><td>00000000000000000000000000000000</td><td>00000000</td></tr>
+<tr><td>1</td><td>00000000000000000000000000000001</td><td>00000001</td></tr>
+<tr><td>2</td><td>00000000000000000000000000000010</td><td>00000010</td></tr>
+<tr><td>3</td><td>00000000000000000000000000000011</td><td>00000011</td></tr>
+<tr><td>127</td><td>00000000000000000000000001111111</td><td>01111111</td></tr>
+<tr><td>128</td><td>00000000000000000000000010000000</td>
+    <td>10000000 00000001</td></tr>
+<tr><td>255</td><td>00000000000000000000000011111111</td>
+    <td>11111111 00000001</td></tr>
+<tr><td>16256</td><td>00000000000000000011111110000000</td>
+    <td>10000000 01111111</td></tr>
+<tr><td>2080768</td><td>00000000000111111100000000000000</td>
+    <td>10000000 10000000 01111111</td></tr>
+<tr><td>266338304</td><td>00001111111000000000000000000000</td>
+    <td>10000000 10000000 10000000 01111111</td></tr>
+<tr><td>4294967295</td><td>11111111111111111111111111111111</td>
+    <td>11111111 11111111 11111111 11111111 00001111</td></tr>
+</table>
+</div>
+
+*Add psuedo code*
+
+*Add reference to proto implementation*
+
+<div class="example">
+```
+zig_zag_encoded_list = [46, 40, 61, 17, 128, 2, 2, 137]
+bytes = [2E 28 3D 11 80 01 02 02 89 01]
+//       └┘ └┘ └┘ └┘ └───┘ └┘ └┘ └───┘
+```
+</div>
+
+### RangeList ### {#rangelist}
+
+A Range is a set of integers i where min <= i <= max. It is represented by
+the pair of integers [min, max], where min <= max and min >= 0.
```

Comment:
Suggest replacing this paragraph with:

A RangeList encodes a set of unsigned 32 bit integers. The set is encoded as a list of disjoint intervals. Each interval is represented by two integers, a start (inclusive) and end (inclusive).

Reply:
done